fvm/transactionContractFunctionInvocator.go

```diff
@@ -44,6 +45,13 @@ func NewTransactionContractFunctionInvoker(
 func (i *TransactionContractFunctionInvoker) Invoke(env Environment, parentTraceSpan opentracing.Span) (cadence.Value, error) {
 	var span opentracing.Span
 
+	if env.StateHolder().EnforceComputationLimits {
+		err := env.StateHolder().State().MeterComputation(meter.ComputationKindContractFunctionInvoke, 1)
+		if err != nil {
+			return nil, err
+		}
+	}
+
```

Comment:
I think instead of having general concept of contractFunctionInvoke, we should meter things where we actually call this Invoke method and capture meter with a more detailed computation type (e.g. Check Stroage, ...)

Reply:
Both is actually metered.

The `meter.ComputationKindContractFunctionInvoke` is meant as the meter for the overhead the ContractFunctionInvoke has. 

The contract function invocation itself will still be using and incrementing the meter on the `env` as the `env` is passed into the invocation.